src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -590,10 +670,8 @@ private void Poll()
                 {
                     // this would return immediately if there are messages waiting inside the client queue buffer
                     var consumed = _consumer.Consume(timeout);
-                    if (consumed is null)
+                    if (consumed is null || _rebalanceInProgress) // bail out if we detect the start of a re-balance
```

Comment:
This would be the most elegant solution to our problem if the Confluent.Kafka client plays nicely with us - as soon as we detect a rebalance is about to start (either by having a partition revoked or lost) we bail out of the polling loop and process whatever results we've already gathered. We'd then wait for the rebalance to complete and for all of the affected stages to send us new `RequestMessages` with the updated partition assignments, which should prevent the error from #415 from occurring in the first place.

Reply:
Given that the events from the Kafka consumer occur synchronously in the same thread as the `Poll` method, I think this will probably work. The only trouble is we might have to keep polling in order to get the `OnAssign` operation to complete.